app/components/channel_list/categories/body/index.ts

```diff
@@ -87,8 +92,8 @@ type EnhanceProps = {category: CategoryModel; locale: string; currentUserId: str
 
 const enhance = withObservables(['category'], ({category, locale, database, currentUserId}: EnhanceProps) => {
     const observedCategory = category.observe();
-    const sortedIds = observedCategory.pipe(
-        switchMap((c) => getSortedIds(database, c, locale)),
+    const sortedChannels = observedCategory.pipe(
```

Comment:
Down the line, we are getting every channel in its own component. This seemed a possible reason why there was some performance degradation. Instead of that, we get all channels here, and just pass the channels instead of the ids. That way there is no need to reach the database several times from different components.